### PR TITLE
#167165024 User Should Be Able To Comment On An Article

### PR DESCRIPTION
**Pull request by Tunji17**
### What does this PR do?
User should be able to comment on articles

#### Description of Task to be completed?
Adds a comment input field
Adds cards for fetched comments
Refactored tests to accommodate change

#### How should this be manually tested?
git fetch origin ft-comment-on-articles-167165024
npm run dev 
navigate to the single article page and scroll down 
you should be able to comment on an article

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#167165024](https://www.pivotaltracker.com/story/show/167165024)

#### Screenshots (if appropriate)
Screenshot 2019-09-10 at 11 41 29 AM

### CODE REVIEW

File: jsconfig.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
       "@Types/*": ["src/redux/actions/types/*"],
       "@Common/*": ["src/components/common/*"],
       "@Views/*": ["src/components/views/*"],
-      "@Utils/*": ["utils/*"],
+      "@Utils/*": ["utils/*"]
```
Comment (mazma1) [inline jsconfig.json:15]: Why did you take out the trailing comma here?

Comment (Tunji17) [reply jsconfig.json:15]: My eslint was flagging it.

File: src/redux/actions/articleActions.js [hunk 2]
```diff
@@ -42,6 +52,18 @@ export const getArticleBySlug = slug => async (dispatch) => {
   }
 };
 
+export const createArticleComment = (slug, comment) => async (dispatch) => {
+  dispatch(articleLoading());
+  try {
+    const token = localStorage.getItem('jwtToken');
+    SetAuthToken(`Bearer ${token}`);
```
Comment (mazma1) [inline src/redux/actions/articleActions.js]: Why is this here?

Comment (Tunji17) [reply src/redux/actions/articleActions.js]: I would refactor it, I was trying to make it work

File: src/redux/actions/articleActions.js [hunk 3]
```diff
@@ -24,6 +26,14 @@ export const getSingleArticle = article => ({
   },
 });
 
+export const createArticleCommentAction = comment => ({
```
Comment (mazma1) [inline src/redux/actions/articleActions.js]: We know this is an action, I don't think you need to append `Action` to the variable name

File: src/components/common/comments/ArticleComment.spec.js [hunk 4]
```diff
@@ -1,11 +1,44 @@
...
+
+  //   //       },
+  //   //     },
+  //   //   },
+  //   //   createdAt: '',
+  //   // };
+  //   const component = shallow(<ArticleCommentFormComponent />);
+
+  //   expect(component.find('button')).toHaveLength(1);
+  // });
```
Comment (mazma1) [inline src/components/common/comments/ArticleComment.spec.js]: Why did you push this? Is it still needed?

File: src/redux/actions/articleActions.js (hunk 3 above)
Comment (Tunji17) [reply src/redux/actions/articleActions.js]: noted

File: src/components/common/comments/ArticleComment.spec.js (hunk 4 above)
Comment (Tunji17) [reply src/components/common/comments/ArticleComment.spec.js]: yes, I was making my test pass, I would uncomment it and write it well

**Comment by mazma1** [comment]
@Tunji17 is your branch up to date with `develop`? I can't view a single article:

Screenshot 2019-09-12 at 2 11 45 PM

**Comment by Tunji17** [comment]
> @Tunji17 is your branch up to date with `develop`? I can't view a single article:
> 
> Screenshot 2019-09-12 at 2 11 45 PM

sorry i'll rebase and push again

**Comment by mazma1** [comment]
@Tunji17 

Screenshot 2019-09-13 at 3 15 41 PM

**Comment by Tunji17** [comment]
> @Tunji17
> 
> Screenshot 2019-09-13 at 3 15 41 PM

sorry it was an oversight, i've pushed the fix

**Comment by mazma1** [comment]
Screenshot 2019-09-13 at 3 53 33 PM

- The text on the button is small, you can increase the size a bit
- I think you can add some space to separate the comment section. A little space between the star icons and "No comments for this article", then between the button and "Related Articles"

**Comment by Tunji17** [comment]
> Screenshot 2019-09-13 at 3 53 33 PM
> 
> * The text on the button is small, you can increase the size a bit
> * I think you can add some space to separate the comment section. A little space between the star icons and "No comments for this article", then between the button and "Related Articles"

okay i'll do that now
